apps/resource_app.go

```diff
@@ -0,0 +1,132 @@
+package apps
+
+import (
+	"context"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/service/apps"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+const (
+	defaultAppProvisionTimeout = 10 * time.Minute
+	deleteCallTimeout          = 10 * time.Second
+)
+
+var appAliasMap = map[string]string{
+	"resources": "resource",
+}
+
+type appStruct struct {
+	apps.App
+}
+
+func (appStruct) Aliases() map[string]map[string]string {
+	return map[string]map[string]string{
+		"apps.appStruct": appAliasMap,
+	}
+}
+
+func (appStruct) CustomizeSchema(s *common.CustomizableSchema) *common.CustomizableSchema {
+
+	// Required fields & validation
+	s.SchemaPath("name").SetRequired().SetForceNew().SetValidateFunc(validation.StringMatch(regexp.MustCompile("^[a-z-]{2,30}$"), "name must contain only lowercase alphanumeric characters and hyphens, and be between 2 and 30 characters long"))
+	// Resources should be a set
+	s.SchemaPath("resource").SetSliceSet()
```

Comment:
Do we gain much from treating this as a set vs a list?

AFAICT this only allows the user to shuffle the order if they want.

Reply:
the diff is also cleaner if the user is removing/adding resources, it will only show the resource being added/removed, rather than the entire list changing.
@alexott is the issue that `slice_set` has some edge cases for reflection? I did some manual testing and it works okay (removing, adding resources)

Reply:
yes, there was a behavior of SDK when doing a diff on structs that contained complex objects, like, arrays or structs, etc. This was caused the way the hash was calculated, etc. Maybe it's not a problem in this case, just wanted to make sure